src/emulator/storage/files.ts

```diff
@@ -390,41 +385,53 @@ export class StorageLayer {
     delimiter: string,
     pageToken: string | undefined,
     maxResults: number | undefined
-  ) {
-    if (!delimiter) {
-      delimiter = "/";
-    }
-
-    if (!prefix) {
-      prefix = "";
-    }
-
-    if (!prefix.endsWith(delimiter)) {
-      prefix += delimiter;
-    }
-
-    let items = [];
+  ): {
+    kind: string;
```

Comment:
Let's drop this `kind` property from the response if it's not used.

Reply:
We want the responses from the storage emulator to match actual GCS:

`curl -i 'https://storage.googleapis.com/storage/v1/b/bookcreator-dev-firetool-tests/o?maxResults=1' `

```
{
  "kind": "storage#objects",
  "nextPageToken": "CgZwcmVmaXg=",
  "items": [
    {
      "kind": "storage#object",
      "id": "bookcreator-dev-firetool-tests/prefix/1628512006128902",
      "selfLink": "https://www.googleapis.com/storage/v1/b/bookcreator-dev-firetool-tests/o/prefix",
      "mediaLink": "https://storage.googleapis.com/download/storage/v1/b/bookcreator-dev-firetool-tests/o/prefix?generation=1628512006128902&alt=media",
      "name": "prefix",
      "bucket": "bookcreator-dev-firetool-tests",
      "generation": "1628512006128902",
      "metageneration": "7",
      "contentType": "application/json",
      "storageClass": "STANDARD",
      "size": "5",
      "md5Hash": "Z0RBlgyhui3gitTlDJ/emA==",
      "crc32c": "t0Xs5w==",
      "etag": "CIaKjf33o/ICEAc=",
      "timeCreated": "2021-08-09T12:26:46.130Z",
      "updated": "2021-08-11T07:17:42.175Z",
      "timeStorageClassUpdated": "2021-08-09T12:26:46.130Z",
      "metadata": {
        "boo": "blah",
        "2firebaseStorageDownloadTokens": "world",
        "firebaseStorageDownloadTokens": "hello"
      }
    }
  ]
}
```

Reply:
Now that `listItems` essentially does the same thing as `listItemsAndPrefixes`, would prefer we collapse the two methods into one and handle the differences in `firebase.ts` and `gcloud.ts`

Reply:
Can you do this in another PR once its merged?